### DISCUSSION TITLE
Wrong timezone is used for default date in some cases

### DISCUSSION THREAD

**Discussion by jfpedroza** [firefly-iii/firefly-iii#7381] [Firefly III] [answered]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I've notice that the default date is incorrect in some cases when it's already next day in UTC (7 pm in my case).

If I open the new transaction page it correctly shows 15/04/2023 but if

1. I clone a transaction, the default date is 16/04/2023
2. Create a bill, the default date is 16/04/2023
3. Try to create a recurrent transaction, the default date is 17/04/2023

### Debug information

Debug information generated at 2023-04-16 00:00:00 Europe/Amsterdam for Firefly III version **6.0.5**.

| Scope | Version |
| --- | --- |
| Firefly III | 6.0.5 |
| Firefly III API | 2.0.1 |
| Build | (unknown), (unknown) |
| DB version | 19 (exp. 19) |
| PHP | `8.2.4` |
| Host | `Linux` |

| System info | Value |
| --- | --- |
| System TZ | Europe/Amsterdam |
| Browser TZ | America/Bogota |
| App environment | local |
| App debug mode | false |
| App cache driver | file |
| App logging | notice, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Max upload | 52428800 (50 MB) |
| Interface | fpm-fcgi |
| Default language | en_US |
| Default locale | equal |
| BCscale | 12 |
| DB drivers | mysql |
| Current driver | mysql |
| Trusted proxies (.env) | `` |

| User info | Value |
| --- | --- |
| Login provider | eloquent  |
| User guard | web  |
| Headers | N/A, N/A  |
| Stateful domains |  |
| Session start | 2023-04-01 00:00:00 |
| Session end | 2023-04-30 23:59:59 |
| Session first | 2022-11-05 00:00:00 |
| User ID | 1 |
| User language | en_US |
| User locale | es_CO |
| Attempt at "es_CO.utf8" | 'es_CO.utf8' |
| Attempt at "es_CO.UTF-8" | 'es_CO.UTF-8' |
| User agent | Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/112.0 |

### Expected behaviour

The default date should still be 15/04 (16/04 in case of recurrent transactions).

### Steps to reproduce

Same steps described above.

### Additional info

The server has UTC timezone but given that default date is set correctly in one case, I would say the expected behavior is that the browser timezone is used. Or at least that the behavior is consistent.

**Comment by JC5** [comment, marked as answer]
Firefly III uses the timezone set in `System TZ`, which in your case is "Europe/Amsterdam". Set the `TZ` environment variable to the correct value and it will work as you expect.

**Reply by jfpedroza** [reply]
It was that. Thanks.

I was not aware of that variable.
